tests/conftest.py

```diff
@@ -74,23 +64,58 @@ def auth_state(MockAuthClient, config, monkeypatch) -> AuthState:
     return auth_state
 
 
-@pytest.fixture
-def duplicate_auth_state(auth_state: AuthState, config) -> AuthState:
-    """This fixture provides an AuthState type object that has the same token value as the
-    auth_state fixture above. This means that caching should make access to either of the
-    returned auth_state objects return values from the same cache as the root key is the
-    token value.
+@pytest.fixture(scope="session", autouse=True)
+def register_api_fixtures():
+    for yaml_file in (pathlib.Path(__file__).parent / "api-fixtures").rglob("*.yaml"):
+        response_set = yaml.safe_load(yaml_file.read_text())
+        register_response_set(yaml_file.stem, response_set)
+
+
+@pytest.fixture(autouse=True)
+def responses() -> responses_module.RequestsMock:
+    """Mock all requests.
 
+    The default `responses.mock` object is returned,
+    which allows tests to access various properties of the mock.
+    For example, they might check the number of intercepted `.calls`.
     """
 
-    # Create a TokenChecker to be used to create a mocked auth_state object
-    checker = TokenChecker(
-        client_id=config["client_id"],
-        client_secret=config["client_secret"],
-        expected_scopes=config["expected_scopes"],
-        expected_audience=config["expected_audience"],
-    )
-    dup_auth_state = checker.check_token(auth_state.bearer_token)
-    # Set up the same mocked Auth client
-    dup_auth_state.auth_client = auth_state.auth_client
-    return dup_auth_state
+    responses_module.reset()
+    responses_module.start()
+    try:
+        yield responses_module.mock
+    finally:
+        responses_module.stop()
+
+
+@pytest.fixture
+def freeze_time() -> t.Generator[
+    t.Callable[[RegisteredResponse], RegisteredResponse], None, None
+]:
+    """Inspect a Globus SDK RegisteredResponse object and freeze time if needed.
+
+    Some responses may only be valid within a specific time range
+    (for example, a token may only be valid within a specific time period).
+    This fixture creates a function that will look for a "freezegun" key
+    in the Response metadata. If found, time will be frozen at that value.
+    """
+
+    frozen_time: t.Optional[freezegun.freeze_time] = None
+
+    def freezer(response: RegisteredResponse) -> RegisteredResponse:
+        """Freeze time based on a "freezegun" key (if any) in the response metadata."""
+
+        if "freezegun" in response.metadata:
+            instant = datetime.datetime.utcfromtimestamp(response.metadata["freezegun"])
+            # Update `frozen_time` in the outer scope.
+            nonlocal frozen_time
+            frozen_time = freezegun.freeze_time(instant)
+            frozen_time.start()
+        return response
```

Comment:
Well, gosh! I give you a blob called `metadata` and look what you do with it!
I think this is a very cool and -- for me -- unanticipated pattern of usage.

Reply:
Haha, hooray! Unanticipated benefits!